<?xml version="1.0"?>
<launch>  
  <node pkg="nodelet" type="nodelet" name="pacman_nodelet_manager" args="manager" output="screen"/>
  
  <node pkg="nodelet" type="nodelet" name="pacman_depth_registry"
        args="load depth_image_proc/register pacman_nodelet_manager"
        output="screen">
    <!-- Topics the nodelet subscribes to: -->
    <remap from="rgb/camera_info" to="/camera/rgb/camera_info" />
    <remap from="depth/camera_info" to="/camera/depth/camera_info" />
    <remap from="depth/image_rect" to="/camera/depth/image_raw" />
    <!-- Topics the nodelet publishes: -->
    <remap from="depth_registered/camera_info" to="/pacman/depth_registered/camera_info" />
    <remap from="depth_registered/image_rect" to="/pacman/depth_registered/image_rect" />
  </node>
  
  <node pkg="nodelet" type="nodelet" name="pacman_depth_pointcloud"
        args="load depth_image_proc/point_cloud_xyz pacman_nodelet_manager"
        output="screen">
    <!-- Topics the nodelet subscribes to: -->
    <!-- Ambiguous. Cold be "/camera/rgb/camera_info or /pacman/depth_registered/camera_info" instead -->
    <remap from="camera_info" to="/pacman/depth_registered/camera_info" />
    <remap from="image_rect" to="/pacman/depth_registered/image_rect" />
    <!-- Topics the nodelet publishes: -->
    <remap from="points" to="/pacman/points" />
  </node>
  
</launch>
